lib/Pipelines/BooleanPipelineRegistration.cpp

```diff
@@ -157,7 +157,11 @@ void registerTosaToBooleanFpgaTfhePipeline(const std::string &yosysFilesPath,
                                   /*abcBooleanGates=*/true);
 
         // Vectorize CGGI operations
-        pm.addPass(cggi::createBooleanVectorizer());
+        if (options.parallelism >= 0) {
```

Comment:
for FPT why is the default (-1) to skip this pass?

Reply:
I was thinking about this. It will turn off the bool parallelism by default and need to be added (parallelism=0) via a flag everytime. @WoutLegiest - are you ok with this change?

An alternative would be to add a separate parallelism option for just the jaxite pipeline.  I can do this as well. wdyt?

Reply:
Would it make more sense for parallelism to default to 0 (infinite parallelism for both)?

Reply:
I would also prefer a default setting of parallelism to 0, i.e. infinite.
I don't really see why we should skip the boolean vectorizer for FPT?

Reply:
The issue with infinite parallelism on TPU is that actually doesnt exist - pmap requires that batch size be equal to the number of local devices or it errors out. (Sorry forgot to mention this pmap quirk)

 So we have 2 options:
1. add the parallelism option to TosaToBooleanTfheOptions : Here parallelism=0 would mean infinite parallelism for fpga, no parallelism for jaxite. Seems a bit counter-intuitive if the same option does different things.
2. TosaToBooleanJaxiteOptions: It wouldn't effect the fpga pipeline at all (assuming the fpga pipeline will almost always prefer infinite parallelism)

lmk what you think. I have a slight preference for option 2.

Reply:
For me both are actually oke. Option two would be the easiest, but generates an additional code.

Is there an option to implement the first option, but setting the parallelism to an acceptable value in the Jaxite pipeline?

Reply:
Definitely agree different behavior is not ideal. I would say you should implement parallelism=0 means max parallelism: infinite parallelism for the FPGA, and then max parallelism for jaxite (retrieved by `jax.local_device_count()`)

Reply:
Alternatively, can you just extend the options class for Jaxite if you just want to use the option in that pipeline?

Reply:
Re: reasonable max value on TPU:  Currently the batch_size is heavily tied to the local device count and errors out if there is a mismatch so default value of no parallelism works better. 

I will go with TosaToBooleanJaxiteOptions extending the exiting options class.

Thanks!